app/src/main/java/com/example/coopertest/TestActivity.kt

```diff
@@ -199,23 +192,14 @@ class TestActivity : AppCompatActivity(), OnMapReadyCallback {
     }
 
     private fun requestPermissions() {
-        ActivityCompat.requestPermissions(
-            this,
-            arrayOf(
-                Manifest.permission.ACCESS_COARSE_LOCATION,
-                Manifest.permission.ACCESS_FINE_LOCATION
-            ),
-            PERMISSION_ID
-        )
+        val intent = Intent(this, LocationActivity::class.java)
+        startActivity(intent)
     }
 
-    override fun onRequestPermissionsResult(
-        requestCode: Int,
-        permissions: Array<String>,
-        grantResults: IntArray
-    ) {
-        if (requestCode == PERMISSION_ID && grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-            requestPermissionsAndLocationUpdates()
-        }
+    override fun onBackPressed() {
```

Comment:
Could you add confirmation for that as well? Like a pop-up with message "Do you really want to quit the test?"